dissect/target/plugins/os/unix/log/journal.py

```diff
@@ -259,16 +262,26 @@
 
 def get_optional(value: str, to_type: Callable):
     """Return the value if True, otherwise return None."""
-    return to_type(value) if value else None
+
+    if not value:
+        return None
+
+    try:
+        return to_type(value)
+
+    except ValueError as e:
+        log.error("Unable to cast '%s' to %s", value, to_type)
+        log.debug("", exc_info=e)
+        return None
 
 
 class JournalFile:
     """Parse Systemd Journal file format.
 
     References:
-        - https://github.com/systemd/systemd/blob/206f0f397edf1144c63a158fb30f496c3e89f256/docs/JOURNAL_FILE_FORMAT.md
-        - https://github.com/libyal/dtformats/blob/c4fc2b8102702c64b58f145971821986bf74e6c0/documentation/Systemd%20journal%20file%20format.asciidoc
-    """  # noqa: E501
+        - https://github.com/systemd/systemd/blob/main/docs/JOURNAL_FILE_FORMAT.md
```

Comment:
Can this be changed to the specific commit that you read to create this PR? Without this it isn't clear which version/commit of the format was used to create these modifications.

Note I am not a reviewer of this PR or an author of this project.

Reply:
Can you also address this? @JSCU-CNI. Using `main` may result in pointing to different/nonexistent documentation in the future.